targets/XPlatCppSdk/source/code/include/playfab/PlayFabAndroidHttpPlugin.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <playfab/PlayFabCallRequestContainer.h>
+#include <playfab/PlayFabPluginManager.h>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// PlayFabAndroidHttpPlugin is an https implementation to interact with PlayFab services using WinHTTP.
+    /// </summary>
+    class PlayFabAndroidHttpPlugin : public IPlayFabHttpPlugin
+    {
+    public:
+        PlayFabAndroidHttpPlugin();
+        PlayFabAndroidHttpPlugin(const PlayFabAndroidHttpPlugin& other) = delete;
+        PlayFabAndroidHttpPlugin(PlayFabAndroidHttpPlugin&& other) = delete;
+        PlayFabAndroidHttpPlugin& operator=(PlayFabAndroidHttpPlugin&& other) = delete;
+        virtual ~PlayFabAndroidHttpPlugin();
+
+        virtual void MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer) override;
+        virtual size_t Update() override;
+
+    protected:
+        struct RequestTask;
+
+        void WorkerThreadEntry();
+        void WorkerThread(void* jniEnv);
+        bool ExecuteRequest(RequestTask& requestTask);
+        bool CheckResponse(RequestTask& requestTask);
+        void SetResponceAsBadRequest(RequestTask& requestTask);
+
+        virtual std::string GetUrl(RequestTask& requestTask) const;
+        virtual void SetPredefinedHeaders(RequestTask& requestTask);
+        virtual void SetHeader(RequestTask& requestTask, const char* name, const char* value);
+        virtual bool GetBinaryPayload(RequestTask& requestTask, void*& payload, size_t& payloadSize) const;
```

Comment:
This code is working on this case, because we are not refer the returned pointer later. the return value is immediatly copied to new memory. And there will be no misuse.
And your recommendation is duplicate memory.
But I totally agree with you this is little gangerous. And if we have to change this, we also have to change another platform too at once.

---
In reply to: [275580593](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r275580593) [](ancestors = 275580593)

Reply:
Suggestion #1 does duplicate memory.  
Suggestion #2 manages the memory. 
Suggestion #3 just basically leaves us in the present state but puts the accessor on the object we are reaching into.

---
In reply to: [275613540](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r275613540) [](ancestors = 275613540,275580593)

Reply:
Still not comfortable with this function design, but I'll resolve it for the moment.


---
In reply to: [276013635](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r276013635) [](ancestors = 276013635,275613540,275580593)